driver/nllloss_driver.hpp

```diff
@@ -0,0 +1,588 @@
+    if(reduction != "none" && reduction != "mean" && reduction != "sum")
+        return miopenStatusInvalidValue;
+
+    input_sizes  = GetInputTensorDimsFromCmd();
+    ignore_index = static_cast<int32_t>(inflags.GetValueInt("ignore_index"));
+
+    std::vector<int> in_len     = input_sizes;
+    std::vector<int> target_len = in_len;
+    target_len.erase(std::next(target_len.begin()));
+    std::vector<int> weight_len = {in_len[1]};
+    std::vector<int> out_len    = target_len;
+
+    auto in_strides     = GetStrides(in_len, 1);
+    auto tar_strides    = GetStrides(target_len, inflags.GetValueInt("contiguous"));
+    auto weight_strides = GetStrides(weight_len, 1);
+
+    SetTensorNd(inputDesc, in_len, in_strides, data_type);
+    SetTensorNd(targetDesc, target_len, tar_strides, data_type);
+    SetTensorNd(weightDesc, weight_len, weight_strides, data_type);
+
+    if(reduction == "none")
+    {
+        divisor             = std::numeric_limits<float>::quiet_NaN();
+        auto output_strides = GetStrides(out_len, 1);
+        SetTensorNd(outputDesc, out_len, output_strides, data_type);
+        SetTensorNd(outputGradDesc, out_len, output_strides, data_type);
+    }
+    else
+    {
+        std::vector<int> out_len_rd = {1};
+        SetTensorNd(outputDesc, out_len_rd, data_type);
+        auto output_strides = GetStrides(out_len_rd, 1);
+        SetTensorNd(outputGradDesc, out_len_rd, output_strides, data_type);
+        if(reduction == "sum")
+            divisor = 1;
+        if(reduction == "mean")
+            divisor = miopen::deref(targetDesc).GetElementSize();
+    }
+
+    SetTensorNd(inputGradDesc, in_len, in_strides, data_type);
+
+    return miopenStatusSuccess;
+}
+
+template <typename Tgpu, typename Tref>
+int NLLLossDriver<Tgpu, Tref>::AddCmdLineArgs()
+{
+    inflags.AddInputFlag("forw", 'F', "1", "Run only Forward NLLLoss (Default=1)", "int");
+    inflags.AddInputFlag("input_dims",
+                         'D',
+                         "16,21,21,21,10",
+                         "The dimensional lengths of the input tensor",
+                         "string");
+    inflags.AddInputFlag("ignore_index", 'g', "-1", "Ignore index", "int");
+    inflags.AddInputFlag("reduce",
+                         'R',
+                         "none",
+                         "Specifies the reduction to apply to the output ('none'|'mean'|'sum') "
+                         "(Default=none to indicate no reduction)",
+                         "string");
+    inflags.AddInputFlag("contiguous",
+                         'c',
+                         "1",
+                         "Is input tensor contiguous? (Default=1 for contiguous tensor)",
+                         "int");
+
+    inflags.AddInputFlag("iter", 'i', "10", "Number of Iterations (Default=10)", "int");
+    inflags.AddInputFlag("verify", 'V', "1", "Verify (Default=1)", "int");
+    inflags.AddInputFlag("time", 't', "1", "Time (Default=0)", "int");
```

Comment:
The default for time is shown as 0, but in reality, it is 1. Please make it consistent.

Reply:
fixed it